src/classes/EP_ManageEPTemplate_CTRL.cls

```diff
@@ -277,6 +277,10 @@ public with sharing class EP_ManageEPTemplate_CTRL {
             for (TaskWrapper wrapper : mapTaskWrappers.values()) {
                 Engagement_Plan_Task__c epTask = wrapper.detail;
                 epTask.Engagement_Plan_Template__c = template.id;
+                if (String.isBlank(epTask.Name)) {
+                    epTask.Name.addError(Label.NameRequired);
```

Comment:
In my testing, addError added a pagemessage, but did not throw an exception, and the save went through. If I threw an exception in addition to addError, there were two pagemessages, so instead I returned null and did the validation before any additional processing, which shows the message both in pagemessages and at the field level.